File: MyTestDemo/module/popwindow/src/main/java/com/xcm91/relation/popwindow/alipay/MainActivity.java
package com.xcm91.relation.popwindow.alipay;

import android.graphics.Rect;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Gravity;
import android.view.View;
import android.widget.Toast;

import com.xcm91.relation.popwindow.R;
import com.xcm91.relation.popwindow.alipay.popwindow.SelectPopupWindow;


public class MainActivity extends AppCompatActivity implements SelectPopupWindow.OnPopWindowClickListener{
    private SelectPopupWindow menuWindow;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_password_test);
    }

    //打开输入密码的对话框
    public void inoutPsw(View view){
        menuWindow = new SelectPopupWindow(this, this);
        Rect rect = new Rect();
        getWindow().getDecorView().getWindowVisibleDisplayFrame(rect);
        int winHeight = getWindow().getDecorView().getHeight();
        menuWindow.showAtLocation(getWindow().getDecorView(), Gravity.BOTTOM, 0, winHeight - rect.bottom);
    }


    @Override
    public void onPopWindowClickListener(String psw, boolean complete, int type) {
        if(complete)
            Toast.makeText(this, "您输入的密码是"+psw, Toast.LENGTH_SHORT).show();
    }
}

File: MyTestDemo/module/kotlin/src/main/java/com/xcm91/relation/kotlin/MainActivity.kt
package com.xcm91.relation.kotlin

import android.app.Activity
import android.os.Bundle
import android.view.Window

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        get1();
        get1(1);

        StringUtils.isEmpty("")


        var mUserInfo = UserInfo()
        mUserInfo.age = 1
        mUserInfo.name="12"
    }

    private fun get1() {


    }


    private fun get1(int1: Int): Int {
        for (i in 0..int1) {

        }

        return int1 + 1
    }


}
